src/create.ts

```diff
@@ -1,9 +1,14 @@
 // Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
 
 import {createContext, useContext, useEffect, useRef, useState} from 'react';
-import {Action, Dispatch, Store} from 'redux';
+import {Action, ActionCreatorsMapObject, Dispatch, Store} from 'redux';
 import shallowEqual from './shallowEqual';
 
+type InferArgs<F> = F extends (...args: infer T) => unknown ? T : never;
```

Comment:
What about handling the special case when the function has no args so that it's not infered as `(...args: any[])`, i.e. allowing any number of args of any type. Something like this comes to my mind:

```ts
type InferArgs<F> =
  F extends () => unknown
    ? never[]
    : F extends (...args: infer T) => unknown
      ? T
      : never;
```

Reply:
Updated and added a test!